### PR TITLE
Setup billing page

### LINKED ISSUE #128: Setup billing page

### PR DESCRIPTION
**Pull request by ArsalanSavand**
Resolves #128 

https://bros.zepel.io/projects/8e893259-4c56-417c-b53b-cf00e4e0c232/lists/405f3ee0-f1c5-4fc0-a617-9b8d962388bd/items/e597d9bb-bf65-4bcb-8e41-d4e24fa3e3a0

### CODE REVIEW

**Comment by ArsalanSavand** [comment]
# Screenshot
image

File: src/assets/i18n/en.json [hunk 1]
```diff
@@ -158,6 +158,15 @@
...
   "SELECT_IMAGE": "Select image",
+  "FEEL_FREE_TO_JOIN_OUR_DISCORD_SERVER": "Feel free to join our new Discord server!",
+  "VALIDATING_PAYMENT": "Your payment is being validated...",
+  "YOUR_PLAN": "Your Plan",
+  "RESTRICTED_PAGE": "You don't have permission to visit this page.",
+  "NO_BILLING": "Your billing will appear here.",
+  "CANCEL_SUBSCRIPTION": "Cancel Subscription",
+  "CANCEL_SUBSCRIPTION_DESCRIPTION": "If you cancel, you no longer have the great features that come with {{ planName }} plan, are you sure ?",
+  "TRANSACTIONS": "Transactions",
+  "SUBSCRIPTION_DETAILS": "Subscription Detail"
```
Comment (Alir3z4) [inline src/assets/i18n/en.json]:
```suggestion
  "SUBSCRIPTION_DETAILS": "Subscription Detail",
```

File: src/index.html [hunk 2]
```diff
@@ -18,6 +18,7 @@
   <meta name="msapplication-TileColor" content="#3498db">
   <meta name="msapplication-config" content="assets/manifest/browserconfig.xml">
   <meta name="theme-color" content="#3498db">
+  <script src="https://widget.cloudpayments.ru/bundles/cloudpayments"></script>
```
Comment (Alir3z4) [inline src/index.html]: Can we load this dynamically and only on the billing tab and not the whole thing ?

Comment (Alir3z4) [reply src/index.html]: I've seen it to be done via https://github.com/gazaret/react-native-cloudpayments, a bridge via react-native.

Maybe we can do the same thing here via Angular as well, just like how it has been done for QuillJs that it has a angular wrapper.

Comment (Alir3z4) [reply src/index.html]: I've found this as well https://stackoverflow.com/questions/34489916/how-to-load-external-scripts-dynamically-in-angular

I think it solves the exact problem we have here.

File: src/app/interfaces/v1/plan.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface Plan {
```
Comment (Alir3z4) [inline src/app/interfaces/v1/plan.ts]: Maybe have some documentation and comments as well, I have no clue what `Plan` is when looking at the code.

File: src/app/interfaces/upgrade-plan.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Plan } from '@app/interfaces/v1/plan';
+
+export interface UpgradePlan {
```
Comment (Alir3z4) [inline src/app/interfaces/upgrade-plan.ts]: Maybe have some documentation and comments as well, I have no clue what `UpgradePlan` is when looking at the code.

File: src/app/interfaces/transaction.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Represents a single transaction information structure
```
Comment (Alir3z4) [inline src/app/interfaces/transaction.ts]: Make the documentation mode clear, what it does and the purpose of it and where it's used.

File: src/app/interfaces/subscription.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { Plan } from '@app/interfaces/v1/plan';
+
+export interface Subscription {
```
Comment (Alir3z4) [inline src/app/interfaces/subscription.ts]: Make the documentation mode clear, what it does and the purpose of it and where it's used.

File: src/app/components/dash/settings/settings-upgrade/settings-upgrade.component.ts [hunk 7]
```diff
@@ -1,15 +1,189 @@
...
+              private settingsUpgradeService: SettingsUpgradeService) {
+  }
+
+  ngOnInit(): void {
+    /**
+     * Get current user
+     */
+    AuthService.user.subscribe((data: UserAuth): UserAuth => this.user = data);
+    /**
+     * Check if owner
```
Comment (Alir3z4) [inline src/app/components/dash/settings/settings-upgrade/settings-upgrade.component.ts]: "Check if owner" what ?

This is not helpful.
Write why you're doing this and what should be achieved.

**Pull request merged**